src/fastcs/backends/rest/rest.py

```diff
@@ -0,0 +1,147 @@
+from collections.abc import Awaitable, Callable, Coroutine
+from dataclasses import dataclass
+from types import MethodType
+from typing import Any
+
+import uvicorn
+from fastapi import FastAPI
+from pydantic import create_model
+
+from fastcs.attributes import AttrR, AttrRW, AttrW, T
+from fastcs.controller import BaseController
+from fastcs.mapping import Mapping
+
+
+@dataclass
+class RestServerOptions:
+    host: str = "localhost"
+    port: int = 8080
+    log_level: str = "info"
+
+
+class RestServer:
+    def __init__(self, mapping: Mapping):
+        self._mapping = mapping
+        self._app = self._create_app()
+
+    def _create_app(self):
+        app = FastAPI()
+        _add_dev_attributes(app, self._mapping)
+        _add_dev_commands(app, self._mapping)
+
+        return app
+
+    def run(self, options: RestServerOptions | None = None) -> None:
+        if options is None:
+            options = RestServerOptions()
+
+        uvicorn.run(
+            self._app,
+            host=options.host,
+            port=options.port,
+            log_level=options.log_level,
+        )
+
+
+def _put_request_body(attribute: AttrW[T]):
+    return create_model(
+        f"Put{str(attribute.datatype.dtype)}Value",
+        **{"value": (attribute.datatype.dtype, ...)},  # type: ignore
+    )
+
+
+def _wrap_attr_put(
+    attribute: AttrW[T],
+) -> Callable[[T], Coroutine[Any, Any, None]]:
+    async def attr_set(request):
+        await attribute.process(request.value)
+
+    # Fast api uses type annotations for validation, schema, conversions
+    attr_set.__annotations__["request"] = _put_request_body(attribute)
+
+    return attr_set
+
+
+def _get_response_body(attribute: AttrR[T]):
+    return create_model(
+        f"Get{str(attribute.datatype.dtype)}Value",
+        **{"value": (attribute.datatype.dtype, ...)},  # type: ignore
+    )
+
+
+def _wrap_attr_get(
+    attribute: AttrR[T],
+) -> Callable[[], Coroutine[Any, Any, Any]]:
+    async def attr_get() -> Any:  # Must be any as response_model is set
+        value = attribute.get()  # type: ignore
+        return {"value": value}
+
+    return attr_get
+
+
+def _add_dev_attributes(app: FastAPI, mapping: Mapping) -> None:
+    for single_mapping in mapping.get_controller_mappings():
+        path = single_mapping.controller.path
+
+        for attr_name, attribute in single_mapping.attributes.items():
+            attr_name = attr_name.title().replace("_", "")
+            d_attr_name = f"{'/'.join(path)}/{attr_name}" if path else attr_name
```

Comment:
Maybe `route` or `path` would be a better name here?

Reply:
I avoided "path" because its already used in this context. I avoided "route" because its not the full route (missing the root `/`) but if it makes it clearer I will add it

Reply:
I have also realised that kebab case seems to be best practice for a restapi - I can do this for the attribute/command level. But controller/subcontroller name I will leave for the developer?